src/command_line_parser.cc

```diff
@@ -1812,7 +1813,17 @@ CLParser::ParseCommandLine(int argc, char** argv)
           params_->filename = optarg;
           break;
         case '?':
-          Usage();
+          if (optopt) {
+            Usage(
+                "Error: Invalid short option: '-" +
+                std::string(1, static_cast<char>(optopt)) + "'");
+          } else if (optind > 1 && argv[optind - 1][0] == '-') {
```

Comment:
Wouldn't you wanna check that the first _two_ characters of `argv[optind - 1]` are `--` to tell user that they passed an invalid long option?

Reply:
That case would be handled by the first conditional clause (optopt would be set). See the screenshots above.

In theory, the check for the dash is not necessary, because in our current code it would always evaluate to true. However, in a future PR, we could refactor the code to parse args that do not have flags (with the current code, they get ignored... e.g. `perf_analyzer -m model fake_arg` would ignore fake_arg). In that case, this extra condition is important.